src/app/modules/main/chat_section/chat_content/messages/module.nim

```diff
@@ -232,9 +236,37 @@ method reevaluateViewLoadingState*(self: Module) =
                        self.firstUnseenMessageState.fetching or
                        self.view.getMessageSearchOngoing())
 
+proc generateAlbumIdToImageMap(self: Module, messages: seq[MessageDto]): Table[string, seq[string]] =
+  var albumIdToImagesMap = initTable[string, seq[string]]()
+
+  for message in messages:
+    if message.albumId in albumIdToImagesMap:
+      albumIdToImagesMap[message.albumId].add(message.image)
+    else:
+      albumIdToImagesMap[message.albumId] = @[message.image]
+  
+  return albumIdToImagesMap
+
+
+proc setQuotedMessageImages(self: Module, messages: seq[MessageDto], items: var seq[Item], albumIdToImagesMap: Table[string, seq[string]]) =
+  for message in messages:
```

Comment:
This look quite suspicious to me that we have `ffor message in messages` here while `setQuotedMessageImages` is already called in the same message loop. You sure everything's ok here? 🙂

Reply:
Good eye. We can probably move `setQuotedMessageImages` out of the loop in `newMessagesLoaded`

Reply:
Let me have a second look on it.